arraycontext/loopy.py

```diff
@@ -41,7 +43,9 @@
 
 _DEFAULT_LOOPY_OPTIONS = lp.Options(
         no_numpy=True,
-        return_dict=True)
+        return_dict=True,
+        enforce_array_accesses_within_bounds="no_check" if sys.flags.optimize
```

Comment:
I'm a bit hesitant about this, because I feel it encourages everyone to run with `-O` all the time, negating the benefit of possible runtime checks.

 I think a possibly better approach might be to just turn off bounds checking for kernels generated by pytato wholesale, as long as no hand-written loopy kernels appear in the DAG.